mautic/mautic
# Full contact plugin

| Q  | A
| --- | ---
| Bug fix? | No
| New feature? | Yes
| Related user documentation PR URL | [140](https://github.com/mautic/documentation/pull/140)
| Related developer documentation PR URL | N/A
| Issues addressed (#s or URLs) | [2905](https://github.com/mautic/mautic/pull/2905), [2971](https://github.com/mautic/mautic/pull/2971)
| BC breaks? | N/A
| Deprecations? | N/A

This PR has the following PRs included:

- [x] #2905 
- [x] #2971

#### Description:
Plugin to pull data from Full Contact via the API about contacts and companies into Mautic.

#### Steps to test this PR:
1. Get a [Full Contact API key](https://portal.fullcontact.com/signup)
1. ~~Apply this PR and the [Company SocialCache PR](https://github.com/mautic/mautic/pull/2971)~~ (already included)
2. Enable the plugin
![image](https://cloud.githubusercontent.com/assets/2924026/20350951/db8137e4-abde-11e6-8c93-7d4561b27f48.png)

3. On the contacts and on the companies pages, there are two buttons: One on the row dropdown menu, and another one on the toolbar at the top:
![image](https://cloud.githubusercontent.com/assets/2924026/20355065/110eec8a-abee-11e6-90f6-c1d6228ed995.png)
![image](https://cloud.githubusercontent.com/assets/2924026/20401066/57993bb0-acc5-11e6-8f3e-16d51cb24c57.png)

4. When clicking one of those buttons a window will pop up to confirm the action:
![image](https://cloud.githubusercontent.com/assets/2924026/20521683/01b85e50-b072-11e6-9da3-fa664fdebc49.png)
![image](https://cloud.githubusercontent.com/assets/2924026/20521701/1ca56712-b072-11e6-868b-8a7806304d20.png)

5. A few moments later, if information was found, the contact/company details will be updated.
